tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/bool/AndStratifiedRule.scala

```diff
@@ -0,0 +1,81 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.bool
+
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.bmcmt.rewriter.ConstSimplifierForSmt
+import at.forsyte.apalache.tla.bmcmt.stratifiedRules.{Rewriter, RewriterScope}
+import at.forsyte.apalache.tla.bmcmt.stratifiedRules.{addCell, StratifiedRule}
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.{BoolT1, OperEx, TlaEx}
+import at.forsyte.apalache.tla.typecomp.TBuilderInstruction
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * Implements the rule for conjunction. Similar to TLC, we short-circuit A /\ B as IF A THEN B ELSE FALSE. This allows
+ * us to introduce an optimization on-the-fly for the conjunctions that were marked with a hint. In this optimization,
+ * we push the context, assume A and check satisfiability of the SMT context. If the context is unsat, we immediately
+ * return FALSE. Otherwise, we pop the context and continue.
```

Comment:
```suggestion
 * Implements the rule for conjunction. Similar to TLC, we short-circuit A /\ B as IF A THEN B ELSE FALSE.
```

This comment is massively outdated. Please update it with a description about whether this is the default behavior or not.